### PR TITLE
CLN: Breakup agg

### PR DESCRIPTION
**Pull request by rhshadrach**
- [ ] closes #xxxx
- [ ] tests added / passed
- [ ] passes `black pandas`
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

Extracted dictionary parts to `agg_dict_like`, and changed from `isinstance(arg, dict)` to `is_dict_like(arg)`. Also renamed `aggregate_multiple_funcs` to `agg_list_like`.

### CODE REVIEW

File: pandas/core/aggregation.py [hunk 1]
```diff
@@ -738,3 +641,99 @@ def aggregate_multiple_funcs(obj, arg, _axis):
...
+            "and transformation operations "
+            "simultaneously"
+        )
+    else:
+        from pandas import Series
+
+        # we have a dict of scalars
+        # GH 36212 use name only if obj is a series
+        if obj.ndim == 1:
+            obj = cast("Series", obj)
```
Comment (ivanovmg) [inline pandas/core/aggregation.py:732]: Is it the same as this?
```
if isinstance(obj, Series):
   ...
```
No need for casting.

File: pandas/core/aggregation.py [hunk 2]
```diff
@@ -738,3 +641,99 @@ def aggregate_multiple_funcs(obj, arg, _axis):
...
+        results = {key: obj._gotitem(key, ndim=1).agg(how) for key, how in arg.items()}
+
+    # set the final keys
+    keys = list(arg.keys())
+
+    # Avoid making two isinstance calls in all and any below
+    is_ndframe = [isinstance(r, ABCNDFrame) for r in results.values()]
+
+    # combine results
+    if all(is_ndframe):
```
Comment (ivanovmg) [inline pandas/core/aggregation.py:713]: In my opinion, would be even better to have this if/elif/else statement in a separate function.
I would not suggest the best name, but I can infer from the comment that something containing ``combine`` could be the way to go.
Although, not a big deal.

File: pandas/core/aggregation.py [hunk 3]
```diff
@@ -667,7 +570,7 @@ def aggregate(obj, arg: AggFuncType, *args, **kwargs):
     return result, True
 
 
-def aggregate_multiple_funcs(obj, arg, _axis):
+def agg_list_like(obj, arg, _axis):
```
Comment (ivanovmg) [inline pandas/core/aggregation.py:573]: Clear function rename!

**Review by ivanovmg — COMMENTED** [review]
Apart from mypy error, LGTM.

File: pandas/core/aggregation.py (hunk 2 above)
Comment (jreback) [reply pandas/core/aggregation.py:713]: yeah this would be nice, this pass or next ok

**Review by jreback — CHANGES_REQUESTED** [review]
can you type the newer functions and add doc-strings , ping on green.

File: pandas/core/aggregation.py (hunk 2 above)
Comment (rhshadrach) [reply pandas/core/aggregation.py:713]: Agreed, and am working in that direction.

File: pandas/core/aggregation.py (hunk 1 above)
Comment (rhshadrach) [reply pandas/core/aggregation.py:732]: See #36677 for a discussion on this section; note that `Series` is no longer imported in this module.

**Comment by rhshadrach** [comment]
Docstrings aren't quite right here; obj can be other types. Hold off on merging.

**Comment by rhshadrach** [comment]
Tests pass when adding the following line to aggregation.aggregate:

    assert isinstance(obj, (Series, DataFrame, SeriesGroupBy, DataFrameGroupBy, BaseWindow, Resampler))

This is a bit ghastly, going to just leave the description as generic unless this specificity is desired.

**Comment by rhshadrach** [comment]
@jreback green-ish

**Review by jreback — APPROVED** [review]

File: pandas/core/aggregation.py [hunk 4]
```diff
@@ -528,133 +528,44 @@ def transform_str_or_callable(
         return func(obj, *args, **kwargs)
 
 
-def aggregate(obj, arg: AggFuncType, *args, **kwargs):
+def aggregate(
+    obj,
```
Comment (jreback) [inline pandas/core/aggregation.py:532]: if you can type this as some point (obj), FrameOrSeries i think.

File: pandas/core/aggregation.py [hunk 5]
```diff
@@ -738,3 +668,118 @@ def aggregate_multiple_funcs(obj, arg, _axis):
                 "cannot combine transform and aggregation operations"
             ) from err
         return result
+
+
+def agg_dict_like(
+    obj,
+    arg: Dict[Label, Union[AggFuncTypeBase, List[AggFuncTypeBase]]],
```
Comment (jreback) [inline pandas/core/aggregation.py:675]: maybe alias this

**Review by jreback — APPROVED** [review]
thanks @rhshadrach some followon suggestion as I know you are tesing this whole module apart

**Pull request merged**